src/store/Providers.tsx

```diff
@@ -8,7 +9,15 @@ type ProvidersProps = {
 }
 
 const Providers: FC<ProvidersProps> = ({ children }) => {
-  return <ThemeProvider attribute="class">{children}</ThemeProvider>
+  const { api, isApiReady } = useApiLoader()
+
+  return (
+    <ThemeProvider attribute="class">
+      {isApiReady && api && (
+        <ApiContext.Provider value={api}>{children}</ApiContext.Provider>
+      )}
+    </ThemeProvider>
```

Comment:
Is ApiContext would have error, the whole application would crash? I mean if it wouldn't, children should be also present even if condition for ApiContext is not met

Reply:
We should replace it with some kind of loading and handle errors. I would simplify it for now